Improve ring documentation (#37181)
home-assistant/home-assistant.github.io: source/_integrations/ring.markdown

The volume sensors are being replaced with the number entity, which allows setting the volume. You should migrate any automations using the volume sensors to the number entity by release 2025.4.0.

### Siren

- Adds a siren entity for every camera that supports a siren. Note the siren will only turn on for 30 seconds before automatically turning off.
- Adds a siren entity for chimes to play the test sound.

### Switch

Once you have enabled the [Ring integration](/integrations/ring), you can start using the switch platform.

- Motion detection - Switches motion detection on and off for cameras.
- In-home chime - Switches on and off a mechanical or digital chime connected to a ring doorbell.

### Light

Once you have enabled the [Ring integration](/integrations/ring), you can start using the light platform. This will add a light for every camera that supports a light (such as a floodlight).

### Number

Once you have enabled the [Ring integration](/integrations/ring), you can start using the number platform.
Currently, it supports showing and setting the volume of the doorbell/chime ring, intercom voice volume, and intercom microphone volume.

## Data updates

> The Ring cloud API is polled for data updates every 60 seconds. When you make changes through Home Assistant (e.g., switching motion detection on), the device's state is updated immediately rather than waiting for the next poll.

Reviewer: ```suggestion
The Ring cloud API is {% term polling polled %} for data updates every 60 seconds. When you make changes through Home Assistant (for example by switching motion detection on), the device's state is updated immediately rather than waiting for the next  {% term polling poll %}.
```

The Ring integration does not connect locally to devices, all communication goes via the cloud.

## Known limitations

### Two-way audio

Two-way audio in camera live view is not currently supported.

### Last recording

To view the last recording entity you will need a Ring subscription.

### Multiple alerts

Some device models send two alerts for a single doorbell ring event.
The integration will provide a workaround for this in a future release.

## Troubleshooting

### Realtime event stability

Home Assistant requires outbound TCP access to port 5228 to connect to Ring's real-time event service.
Ensure your firewall and network configuration allow this connection.

Below are steps to follow if realtime events are not working.

#### Step 1

Issues with Ring alerts may be caused by having too many authenticated devices on your Ring account. Before version 2023.12.0, the Home Assistant Ring integration would register a new entry in `Authorized Client Devices` in the `Control Center` at [ring.com](https://account.ring.com/account/control-center/authorized-devices) on every restart.

> {% warning %}

Reviewer: ```suggestion

{% important %}
```

When cleaning up devices:

1. Only delete entries that start with `ring-doorbell:HomeAssistant` or `Python`.
2. Do NOT delete entries for your phones or other Ring apps.

> 3. If there are too many devices to delete individually, you can use the `Remove all devices` option, but you'll need to re-authorize all your devices afterward.

Reviewer: ```suggestion
3. If there are too many devices to delete individually, you can use the **Remove all devices** option, but you'll need to re-authorize all your devices afterward.
```

> {% endwarning %}

Reviewer: ```suggestion
{% endimportant %}
```

#### Step 2

If you're still experiencing issues after Step 1, try generating a new unique ID for the Home Assistant Ring integration instance.

> To do this, click the three-dot menu on the integration entry and select the `Reconfigure` option.

Reviewer: ```suggestion
To do this, select the three dots {% icon "mdi:dots-vertical" %} menu on the integration entry and select the **Reconfigure** option.
```

Do not try this step before clearing down all the excess `Authorized Client Devices` as per Step 1, or it will simply invalidate the reconfigured entry.

#### Step 3

If alerts are still not working after Steps 1 and 2, try toggling the Motion Warning setting:

1. Go to [ring.com](https://ring.com) and sign in.
2. Select your device.

> 3. Navigate to Device Settings.

Reviewer: ```suggestion
3. Navigate to **Device Settings**.
```

> 4. Find the Motion Warning toggle.

Reviewer: ```suggestion
4. Find the **Motion Warning** toggle.
```

> 5. Turn it off, wait 30 seconds.

Reviewer: ```suggestion
5. Turn it off and wait for 30 seconds.
```

6. Turn it back on.

This has successfully restored alerts for many users.

## Examples

### Automation ideas

- Turn on motion detection for internal cameras when you leave home (with geofencing) and turn off when you get home.
- Start a live feed on a device when the doorbell rings.
- Turn up the volume on a digital chime when you are in the garden.

### Setting up doorbell alerts

You can set an automation up in the Home Assistant UI.

> 1. Find the correct `event` entity under `Entity triggers`.
> 2. For `From` choose the setting `Any state (ignoring attribute changes)`.
> 3. Then add a `Send notification` action under `Notifications`.

Reviewer: ```suggestion
1. Find the correct **event** entity under **Entity triggers**.
2. For **From** choose the setting **Any state (ignoring attribute changes)**.
3. Then add a **Send notification** action under **Notifications**.
```
